examples/mcts.dx

```diff
@@ -0,0 +1,323 @@
+' ## Monte Carlo Tree Search
```

Comment:
Please use a top-level Markdown heading as the title; that way this will appear in the [InDex](https://google-research.github.io/dex-lang/index.html) of Dex examples.
```suggestion
'# Monte Carlo Tree Search
```

Might be aesthetic to change the sub-headings to level 2 to match, also.

Reply:
Done